k8s/object.bzl

```diff
@@ -28,179 +28,186 @@ load(
-
-  if namespace_arg:
-    namespace_arg = "--namespace=\"" +  namespace_arg + "\""
-
-  kubectl_tool_info = ctx.toolchains["@io_bazel_rules_k8s//toolchains/kubectl:toolchain_type"].kubectlinfo
-
-  substitutions = {
-      "%{kubectl_tool}": kubectl_tool_info.tool_path,
-      "%{cluster}": cluster_arg,
-      "%{context}": context_arg,
-      "%{user}": user_arg,
-      "%{namespace_arg}": namespace_arg,
-      "%{kind}": ctx.attr.kind,
-  }
-
-  if hasattr(ctx.executable, "resolved"):
-    substitutions["%{resolve_script}"] = _runfiles(ctx, ctx.executable.resolved)
-    files += [ctx.executable.resolved]
-    files += list(ctx.attr.resolved.default_runfiles.files)
-
-  if hasattr(ctx.executable, "reversed"):
-    substitutions["%{reverse_script}"] = _runfiles(ctx, ctx.executable.reversed)
-    files += [ctx.executable.reversed]
-    files += list(ctx.attr.reversed.default_runfiles.files)
-
-  if hasattr(ctx.files, "unresolved"):
-    substitutions["%{unresolved}"] = _runfiles(ctx, ctx.file.unresolved)
-    files += ctx.files.unresolved
-
-  ctx.actions.expand_template(
-      template = ctx.file._template,
-      substitutions = substitutions,
-      output = ctx.outputs.executable,
-  )
-  return struct(runfiles = ctx.runfiles(files = files))
+    files = [ctx.executable.resolver]
+
+    cluster_arg = ctx.attr.cluster
+    cluster_arg = ctx.expand_make_variables("cluster", cluster_arg, {})
+    if "{" in ctx.attr.cluster:
+        cluster_file = ctx.new_file(ctx.label.name + ".cluster-name")
+        _resolve(ctx, ctx.attr.cluster, cluster_file)
+        cluster_arg = "$(cat %s)" % _runfiles(ctx, cluster_file)
+        files += [cluster_file]
+
+    context_arg = ctx.attr.context
+    context_arg = ctx.expand_make_variables("context", context_arg, {})
+    if "{" in ctx.attr.context:
+        context_file = ctx.new_file(ctx.label.name + ".context-name")
+        _resolve(ctx, ctx.attr.context, context_file)
+        context_arg = "$(cat %s)" % _runfiles(ctx, context_file)
+        files += [context_file]
+
+    user_arg = ctx.attr.user
+    user_arg = ctx.expand_make_variables("user", user_arg, {})
+    if "{" in ctx.attr.user:
+        user_file = ctx.new_file(ctx.label.name + ".user-name")
+        _resolve(ctx, ctx.attr.user, user_file)
+        user_arg = "$(cat %s)" % _runfiles(ctx, user_file)
+        files += [user_file]
+
+    namespace_arg = ctx.attr.namespace
+    namespace_arg = ctx.expand_make_variables("namespace", namespace_arg, {})
+    if "{" in ctx.attr.namespace:
+        namespace_file = ctx.new_file(ctx.label.name + ".namespace-name")
+        _resolve(ctx, ctx.attr.namespace, namespace_file)
+        namespace_arg = "$(cat %s)" % _runfiles(ctx, namespace_file)
+        files += [namespace_file]
+
+    if namespace_arg:
+        namespace_arg = "--namespace=\"" + namespace_arg + "\""
+
+    kubectl_tool_info = ctx.toolchains["@io_bazel_rules_k8s//toolchains/kubectl:toolchain_type"].kubectlinfo
```

Comment:
Note this is the only thing that changed in this file, but buildifier decided to change lots of other things

Reply:
Holy crap Batman

Reply:
Why don’t you run buildifier as a separate PR?

Reply:
ugh, I think we have a check for some files to be properly formatted (or thought) and ran buildifier just to be on the safe side. Created https://github.com/bazelbuild/rules_k8s/pull/230 to just do buildifier; will rebase this once that is in. No need to review this PR until I have that PR in and the e2e tests passing. Thanks!